### PR TITLE
Rewriting project in typescript

### PR DESCRIPTION
**Pull request by ReversableCode**
I have rewritten the project in typescript and added the necessary types in next-env.d.ts

Can you take a look and see if everything still works? I tested it with my own credentials and It's working fine.

### CODE REVIEW

**Review by AlaaZorkane — CHANGES_REQUESTED** [review]
I think `middlewares/42.tsx` could be `middlewares/42.ts` since it uses no JSX? What do you think?

File: components/cards/Profile.tsx [hunk 1]
```diff
@@ -31,7 +33,7 @@ export default function Profile({
         <style>
           {`.slideUp{animation-duration:1s;animation-name:slideUp}.slideDown{animation-duration:1s;animation-name:slideDown}.flipInX{animation-duration:1s;animation-name:flipInX}@keyframes slideUp{from{opacity:0;transform:translateY(20px)}25%{opacity:0}to{opacity:1;transform:translateY(0)}}@keyframes slideDown{from{opacity:0;transform:translateY(-20px)}25%{opacity:0}to{opacity:1;transform:translateY(0)}}@keyframes flipInX{from{transform:perspective(400px) rotate3d(1,0,0,90deg);animation-timing-function:ease-in;opacity:0}40%{transform:perspective(400px) rotate3d(1,0,0,-20deg);animation-timing-function:ease-in}60%{transform:perspective(400px) rotate3d(1,0,0,10deg);opacity:1}80%{transform:perspective(400px) rotate3d(1,0,0,-5deg)}to{transform:perspective(400px)}}`}
         </style>
-        <div xmlns="http://www.w3.org/1999/xhtml">
```
Comment (mohouyizme) [inline components/cards/Profile.tsx:34]: This was done on purpose, it's a trick to render HTML inside an SVG in a GitHub `.md` file.
Based on: https://github.com/sindresorhus/css-in-readme-like-wat

**Comment by ReversableCode** [comment]
@AlaaZorkane True. I'll change it.
Thanks for the helpful observation!

File: components/cards/Profile.tsx [hunk 2]
```diff
@@ -1,5 +1,8 @@
-export default function Profile({
-  user: {
+import { type } from "os"
```
Comment (mohouyizme) [inline components/cards/Profile.tsx:1]: What's that for?

File: next-env.d.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+/// <reference types="next" />
+/// <reference types="next/types/global" />
+
+interface FortyTwoCursusData {
```
Comment (mohouyizme) [inline next-env.d.ts:4]: The file needs to be formatted base on `.prettierrc` config, you can use `Prettier` extension for `vscode`.

**Review by mohouyizme — CHANGES_REQUESTED** [review]
Thank you for the pull request 🙏! although there are some points to address before merging, I added comments on stuff that need changes.
Also, there is an issue related to styling that I couldn't find the source of it yet, check this testing deploy: https://1337-readme-5h3zst96a.vercel.app/

Thank you again 🙏!

File: middlewares/42.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import axios from 'axios'
```
[the remark this answers is not in the document]
Comment (mohouyizme) [reply middlewares/42.tsx]: Apparently it's changed!

File: .all-contributorsrc [hunk 5]
```diff
@@ -31,6 +31,15 @@
         "code",
         "ideas"
       ]
+    },
+    {
```
[the remark this answers is not in the document]
Comment (ReversableCode) [reply .all-contributorsrc:35]: I realized that after submitting the pull request. I'll remove it right away!

File: middlewares/42.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ReversableCode) [reply middlewares/42.tsx]: Yes. I did fix it afterwards

File: package.json [hunk 6]
```diff
@@ -10,6 +10,10 @@
     "build": "next build",
     "start": "next start"
   },
+  "contributors": [
```
[the remark this answers is not in the document]
Comment (ReversableCode) [reply package.json:13]: I realized that after submitting the pull request. I'll remove it right away!

File: components/cards/Profile.tsx (hunk 1 above)
Comment (ReversableCode) [reply components/cards/Profile.tsx:34]: Typescript was not happy with that so I changed div tag to svg instead

File: components/cards/Profile.tsx (hunk 2 above)
Comment (ReversableCode) [reply components/cards/Profile.tsx:1]: That my IDE being confused 😅

File: next-env.d.ts (hunk 3 above)
Comment (ReversableCode) [reply next-env.d.ts:4]: Sorry I forgot to lint before submitting the merge request

File: components/cards/Profile.tsx (hunk 1 above)
Comment (mohouyizme) [reply components/cards/Profile.tsx:34]: I see!
I did a test and it seems that it's working fine inside a GitHub `README.md` file, you can keep it!

**Comment by ReversableCode** [comment]
@mohouyizme the styling issue is related to tailwindcss's config as it's configured to purge js files instead to tsx files. I rolled up the changes. Can you take a look at them.

**Review by AlaaZorkane — APPROVED** [review]

**Comment by mohouyizme** [comment]
@all-contributors please add @ReversableCode for code code

**Comment by mohouyizme** [comment]
@ReversableCode Thank you so much for contributing!
well done 🎉!

**Comment by mohouyizme** [comment]
I've merged the pull request to `develop` branch for the next release! meanwhile, I added you to the list of contributors!

**Comment by mohouyizme** [comment]
Thanks @AlaaZorkane 🙏!

**Pull request merged**